sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelXTests.cs

```diff
@@ -37,15 +39,20 @@ protected override void CompareModels(ModelX model, ModelX model2, string format
 
         protected override string GetExpectedResult(string format)
         {
-            string expected = "{\"kind\":\"X\",\"name\":\"xmodel\"";
-            expected += ",\"fields\":[\"testField\"]";
-            expected += ",\"keyValuePairs\":{\"color\":\"red\"}";
+            var builder = new StringBuilder();
+            builder.Append("{\"fields\":[\"testField\"]")
+                .Append(",\"keyValuePairs\":{\"color\":\"red\"}");
             if (format == "J")
-                expected += ",\"xProperty\":100";
+            {
+                builder.Append(",\"xProperty\":100");
+            }
+            builder.Append(",\"kind\":\"X\",\"name\":\"xmodel\"");
             if (format == "J")
-                expected += ",\"extra\":\"stuff\"";
-            expected += "}";
-            return expected;
+            {
+                builder.Append(",\"extra\":\"stuff\"");
+            }
+            builder.Append("}");
+            return builder.ToString();
```

Comment:
I only changed the order of properties in the expected string.
We have to change this because the generator generates serialization of properties in a certain and deterministic order (especially when there is a base model)